packages/storage-plugin/src/storage.plugin.ts

```diff
@@ -57,41 +55,42 @@ export class NgxsStoragePlugin implements NgxsPlugin {
         }
 
         const isMaster = key === DEFAULT_STATE_KEY;
-        let val: any = this._engine.getItem(key!);
-
-        if (val !== 'undefined' && val != null) {
-          try {
-            const newVal = this._options.deserialize!(val);
-            val = this._options.afterDeserialize!(newVal, key);
-          } catch (e) {
-            // Caretaker note: we have still left the `typeof` condition in order to avoid
-            // creating a breaking change for projects that still use the View Engine.
-            if (typeof ngDevMode === 'undefined' || ngDevMode) {
-              console.error(
-                `Error ocurred while deserializing the ${key} store value, falling back to empty object, the value obtained from the store: `,
-                val
-              );
-            }
-            val = {};
+        let val: any;
+        try {
+          val = this._engine.getItem(key!);
+          if (val === 'undefined' || val == null) {
+            continue;
           }
-
-          if (this._options.migrations) {
-            this._options.migrations.forEach(strategy => {
-              const versionMatch =
-                strategy.version === getValue(val, strategy.versionKey || 'version');
-              const keyMatch = (!strategy.key && isMaster) || strategy.key === key;
-              if (versionMatch && keyMatch) {
-                val = strategy.migrate(val);
-                hasMigration = true;
-              }
-            });
+          const newVal = this._options.deserialize!(val);
+          val = this._options.afterDeserialize!(newVal, key);
+        } catch (e) {
+          // Caretaker note: we have still left the `typeof` condition in order to avoid
+          // creating a breaking change for projects that still use the View Engine.
+          if (typeof ngDevMode === 'undefined' || ngDevMode) {
+            console.error(
+              `Error ocurred while deserializing the ${key} store value, falling back to default store value, the value obtained from the store: `,
+              val
+            );
           }
+          continue;
```

Comment:
We don't want store an empty object when this doesn't work, it will break the default state values, so we just ignore this part.